nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/util/SFTPTransfer.java

```diff
@@ -340,7 +356,7 @@ protected void getListing(final String path, final int depth, final int maxResul
 
                 if (isIncludedFile(entry, symlink) && (filteringDisabled || pathMatched)) {
                     if (filteringDisabled || fileFilterPattern == null || fileFilterPattern.matcher(entryFilename).matches()) {
-                        listing.add(newFileInfo(entry, path));
+                        listing.add(newFileInfo(path, entry.getName(), entry.getAttributes()));
```

Comment:
This change does not account for the possibility of `entry` being `null` as implied by the previous behavior.

Reply:
In this particular context, the `RemoteResourceInfo` may not return directory status if the path is a symbolic link.

Reply:
Thank you for your feedback @exceptionfactory.

The call to `newFileInfo` is preceded by a call to `isIncludedFile` inside the if expression, which in turn calls `.isRegularFile()` on the `RemoteResourceInfo entry` passed in. 
In case the `entry` was `null`, a `NullPointerException` would be thrown before reaching `newFileInfo`. 

Thus I figured the null check is not required in `newFileInfo`, but I can add an explicit `null` check into the if clause, if that's desired.

Reply:
Very good point @EndzeitBegins, on closer inspection, I agree with you that the null check there should be unnecessary given that it would have failed in `isIncludedFile()` based on current behavior.

Reply:
Regarding the second question, it appears that the behavior remains the same. Thanks for the helpful reply, this looks good.